components/ReplySearchItem.js

```diff
@@ -0,0 +1,203 @@
+    margin: '12px 0',
+    flex: 1,
+  },
+  divider: {
+    margin: `${theme.spacing(2)}px 0`,
+    background: theme.palette.secondary[100],
+  },
+  highlight: {
+    color: theme.palette.primary[500],
+  },
+  openModalButton: {
+    position: 'relative',
+    width: '100%',
+    border: 'none',
+    outline: 'none',
+    cursor: 'pointer',
+    background: theme.palette.secondary[100],
+    color: theme.palette.secondary[300],
+    padding: theme.spacing(2),
+    borderBottomLeftRadius: 8,
+    borderBottomRightRadius: 8,
+  },
+  view: {
+    display: 'none',
+    position: 'absolute',
+    right: theme.spacing(2),
+    '& > svg': {
+      verticalAlign: 'middle',
+      marginLeft: theme.spacing(1),
+    },
+    [theme.breakpoints.up('md')]: {
+      display: 'inline',
+    },
+  },
+  article: {
+    padding: 0,
+    borderRadius: 0,
+    '&:not(:last-child)': {
+      paddingBottom: 12,
+      marginBottom: theme.spacing(3),
+      borderBottom: `1px solid ${theme.palette.secondary[200]}`,
+    },
+  },
+}));
+
+export default function ReplySearchItem({
+  id,
+  articleReplies = [],
+  query = '',
+  ...reply
+}) {
+  const articleReply = articleReplies[0];
+  if (!articleReply) return null;
+
+  const [open, setOpen] = useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const replyCount = articleReplies.length;
+
+  const {
+    article,
+    feedbacks,
+    positiveFeedbackCount,
+    negativeFeedbackCount,
+    ownVote,
+  } = articleReply || {};
+
+  const classes = useStyles();
+
+  return (
+    <li className={classes.root}>
+      <Box p={{ xs: 2, md: 4.5 }}>
+        <ArticleInfo article={article} />
+        <div className={classes.flex}>
+          <ExpandableText className={classes.content} lineClamp={3}>
+            {nl2br(article.text)}
+          </ExpandableText>
+        </div>
+        <Divider classes={{ root: classes.divider }} />
+
+        <ReplyItem
+          key={id}
+          replyId={reply.id}
+          reply={reply}
+          feedbacks={feedbacks}
+          positiveFeedbackCount={positiveFeedbackCount}
+          negativeFeedbackCount={negativeFeedbackCount}
+          ownVote={ownVote}
+          query={query}
+        />
+      </Box>
+      <button
+        type="button"
+        className={classes.openModalButton}
+        onClick={handleClickOpen}
+      >
+        {ngettext(
+          msgid`This reply is also used in ${replyCount} other message`,
+          `This reply is also used in ${replyCount} other messages`,
+          replyCount
+        )}
+        <span className={classes.view}>
+          {t`view`}
+          <VisibilityIcon />
+        </span>
+      </button>
+      <Dialog open={open} onClose={handleClose}>
+        <DialogTitle>{t`This reply is used in following messages`}</DialogTitle>
+        <DialogContent>
+          <PlainList>
+            {articleReplies.map(({ article }) => (
```

Comment:
Since this button is supposed to show "other" messages, we should not repeat the article (first original article) shown outside of the dialog.

Please filter out the article that is already shown above.

<img width="1192" alt="螢幕快照 2020-06-01 下午2 09 28" src="https://user-images.githubusercontent.com/108608/83380873-b36ebe00-a411-11ea-9aa1-080d0e5846f0.png">